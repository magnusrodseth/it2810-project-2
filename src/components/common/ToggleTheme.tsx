import { useContext } from 'react'
import { Toggle } from 'react-daisyui'
import { AppContext } from '../../state/context'

const ToggleTheme = () => {
  const { theme, toggleTheme } = useContext(AppContext)
  const isDark = theme === 'dark'

  return (
    <div className="flex justify-center items-center space-x-2 mx-4">
      <Toggle color="secondary" onChange={toggleTheme} checked={!isDark} />
      <span data-testid="toggle-label">{isDark ? 'Dark' : 'Light'}</span>
    </div>
  )
}

export default ToggleTheme
