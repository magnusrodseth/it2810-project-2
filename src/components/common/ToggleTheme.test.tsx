import renderer from 'react-test-renderer'
import ToggleTheme from './ToggleTheme'
import { render, screen } from '@testing-library/react'

describe('ToggleTheme', () => {
  it('renders correctly', () => {
    const component = renderer.create(<ToggleTheme />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders the correct initial theme', () => {
    render(<ToggleTheme />)
    const toggle = screen.getByRole('checkbox')
    expect(toggle).not.toBeChecked()
  })
})

export { }
