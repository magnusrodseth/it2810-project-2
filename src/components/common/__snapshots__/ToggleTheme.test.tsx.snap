// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleTheme renders correctly 1`] = `
<div
  className="flex justify-center items-center space-x-2 mx-4"
>
  <input
    checked={false}
    className="toggle toggle-secondary"
    onChange={[Function]}
    type="checkbox"
  />
  <span
    data-testid="toggle-label"
  >
    Dark
  </span>
</div>
`;
